src/Tizen.NUI.Components/Controls/ItemsView/LinearLayouter.cs

```diff
@@ -0,0 +1,439 @@
+        private List<float> ItemPosition = new List<float>();
+        private List<float> ItemSize = new List<float>();
+        private int ItemSizeChanged = -1;
+        private CollectionView colView; 
+
+        private float GetItemSize(int index)
+        {
+            if (colView.SizingStrategy == ItemSizingStrategy.MeasureAll)
+            {
+                return ItemSize[index];
+            }
+            else
+            {
+                return StepCandidate;
+            }
+            
+        }
+        private void UpdatePosition(int Index)
+        {
+            bool IsGroup = (colView.InternalItemSource is IGroupableItemSource);
+            bool IsGroupHeader = false;
+            bool IsGroupFooter = false;
+
+            if (Index <= 0) return;
+            if (Index >= colView.InternalItemSource.Count)
+
+            if (IsGroup)
+            {
+                IsGroupHeader = (colView.InternalItemSource as IGroupableItemSource).IsGroupHeader(Index);
+                IsGroupFooter = (colView.InternalItemSource as IGroupableItemSource).IsGroupFooter(Index);
+                //Do Something
+            }
+
+            ItemPosition[Index] = ItemPosition[Index-1] + GetItemSize(Index);           
+        }
+
+        private void FindVisibleItems(Vector2 visibleArea)
+        {
+            int MaxIndex = colView.InternalItemSource.Count - 1;
+            if (colView.SizingStrategy == ItemSizingStrategy.MeasureAll)
+            {
+                //If Position is exist, need to find proper visible item!
+                FirstVisible = Convert.ToInt32(Math.Abs(visibleArea.X / StepCandidate)) - 3;
+                LastVisible = Convert.ToInt32(Math.Abs(visibleArea.Y / StepCandidate)) + 3;
+            }
+            else
+            {
+                //Need to Consider GroupHeight!!!!
+                FirstVisible = Convert.ToInt32(Math.Abs(visibleArea.X / StepCandidate)) - 3;
+                LastVisible = Convert.ToInt32(Math.Abs(visibleArea.Y / StepCandidate)) + 3;
+            }
+            if (FirstVisible < 0) FirstVisible = 0;
+            if (LastVisible > (MaxIndex)) LastVisible = MaxIndex;
+        }
+
+        private ViewItem GetVisibleItem(int index)
+        {
+            foreach (ViewItem item in VisibleItems)
+            {
+                if (item.Index == index) return item;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Clean up ItemsLayouter.
+        /// </summary>
+        /// <param name="view"> ItemsView of layouter. </param>
+        /// <since_tizen> 8 </since_tizen>
+        /// This may be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void Initialize(ItemsView view)
+        {
+            colView = view as CollectionView;
+            if (colView == null)
+            {
+                throw new NotSupportedException();
```

Comment:
`ArgumentException`이 적절해 보입니다.
```suggestion
                throw new ArgumentException(["error message"], nameof(view));
```